### ISSUE TITLE
Performance issues?

### ISSUE THREAD

**Issue by obiwarn** [AllenFang/react-bootstrap-table#695]
This is a minor issue but when I do a reactive search in my table, Griddle felt a lot snappier than this table (I guess it could have to do with column resizing?). Maybe there is room for performance improvements (e.g. fixed widths or sth.)?

**Comment by obiwarn** [comment]
Um. Just wanted to let you know this issue gets quite massive when you have a decent table. e.g. mine has like 10 columns with 3 dates and when I want to show more than 10 rows it gets quite laggy.

**Comment by AllenFang** [comment]
@obiwarn, yap, there's actually a known performance issue. I'll try to fix it. But really need some time.. Sorry, I'm also busy on my work. So PR is welcome.

**Comment by zekedroid** [comment]
By fixing it, are you talking about implementing virtual rendering? Because that would be awesome in many levels. It's the last thing this needs to totally overshadow other static table libraries.

**Comment by obiwarn** [comment]
@AllenFang indeed. I reviewed all popular table libs and yours is just fantastic and has such great code samples.

**Comment by AllenFang** [comment]
@obiwarn, thanks, I'll do my best. BTW, #566 tried to fix the rendering performance issue and I still consider if add this PR to ```v3.0.0```. Anyway, there're a lots of jobs to do in ```v3.0.0```, so PR is welcome :)

**Comment by oakley808** [comment]
I'd have to agree with others here. This is by far the best table component I've used. 👏 
I ditched fixed-data-table and griddle for react-bootstrap-table. But the performance on 750 rows X 25 columns is about 10s to render. [ouch]  But I can use pagination to mitigate it for now. :-)

**Comment by oeph** [comment]
@AllenFang i implemented lazy loading for this table, which should fix the performance issue. I can provide a PR, but this will take some time since i have a lot to do at work.

**Comment by AllenFang** [comment]
@oeph, that's ok, PR is welcome :)

**Comment by kzbikowski** [comment]
@oeph any progress in terms of lazy loading? I'm thinking of implementing it but don't want to duplicate the work

**Comment by oeph** [comment]
@kzbikowski i have it done. i will merge on current branch and make a PR. not sure tho on the exact time

**Comment by SimonDKnight** [comment]
Hi @oeph , did your implementation of infinite scroll get merged in? I am super keen to get infinite scroll working on the table. Thanks. Simon

**Comment by oeph** [comment]
since the lazy loading was implemented last year nov. there are major changes within the component itself. I'm working on the solution in the current version without breaking the current API, but for now i would suggest to use the method described in #1133 

since i'm busy at work i can't make any promises when i will have this finished

**Comment by sidewayskumara** [comment]
Hi just wanted to ask if lazy loading/virtualization is still being actively worked on? If not, in the above comment it @oeph  mentions using this library with react-virtualized. Wondering if anyone could provide a simple example of how that would work. Thanks for this great library!

**Comment by JohnVLinton** [comment]
I'm having a lot of issues too with table load time.  I have basically two tables on screen, one hidden, then when a user hits a tab, that table is then displayed.  I'd be happy to do something more like div display:block vs. none to quickly show/hide it but doesn't seem obvious how I do this in React, so each tab toggle requires a real-time redraw...

**Comment by oeph** [comment]
@JohnVLinton To me that sounds more like a problem with the tab component. It could just render the selected tab instead of rendering both and using display:none;

Since react-bootstrap-table is not actively worked on anymore, you should maybe look at https://github.com/react-bootstrap-table/react-bootstrap-table2
